src/host/ft_host/API_OutputTests.cpp

```diff
@@ -159,15 +159,16 @@ void OutputTests::WriteConsoleOutputWOutsideBuffer()
     // Call the API and confirm results.
 
     //  move outside in X and Y directions
-    auto shiftedRegion = region;
-    shiftedRegion.Left += bufferSize.X;
-    shiftedRegion.Right += bufferSize.X;
-    shiftedRegion.Top += bufferSize.Y;
-    shiftedRegion.Bottom += bufferSize.Y;
-
-    auto affected = shiftedRegion;
+    auto affected = region;
+    affected.Left += bufferSize.X;
+    affected.Right += bufferSize.X;
+    affected.Top += bufferSize.Y;
+    affected.Bottom += bufferSize.Y;
+    auto expected = affected;
+    expected.Right = expected.Left - 1;
+    expected.Bottom = expected.Top - 1;
     VERIFY_WIN32_BOOL_SUCCEEDED(WriteConsoleOutputW(consoleOutputHandle, buffer.data(), regionDimensions, regionOrigin, &affected));
-    VERIFY_ARE_EQUAL(shiftedRegion, affected);
+    VERIFY_ARE_EQUAL(expected, affected);

```

Comment:
the actual expected values have changed here. does this have an impact on API consumers?

Reply:
To quote the above MSDN link again:
> If the rectangle specified by `lpWriteRegion` lies completely outside the boundaries of the console screen buffer, [...] no data is written. In this case, the function returns [...] such that the `Right` member is less than the `Left`, or the `Bottom` member is less than the `Top`.

Meanwhile this test asserted that we return the rectangle completely unmodified. It's caused by this branch:
https://github.com/microsoft/terminal/blob/ec9289288e6581b8d1cbb5f7d334422de4dee07a/src/host/directio.cpp#L622-L628